include/aero/aero.h.in

```diff
@@ -6,11 +6,35 @@
 
 #include <assert.h>
 #include <float.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <math.h>
 
 /// Floating point type (with proper precision).
 typedef @AERO_REAL_TYPE@ aero_real_t;
 
 /// Machine-precision "epsilon" for select floating point type.
 static const aero_real_t aero_epsilon = @AERO_C_EPSILON@;
 
+/// Compares floating-point numbers
+///
+/// If the absolute relative difference between the numbers is within a
+/// tolerance based on the precision of \c aero_real_t, the function returns
+/// true. Otherwise, it returns false.
+static bool almost_equal( const aero_real_t a, const aero_real_t b ) {
+  if (a == 0.0 && b == 0.0) return true;
+  if (fabs(a - b) * 2.0 / fabs(a + b) < 5.0 * aero_epsilon ) return true;
+  return false;
+}
+
+/// Compares floating-point numbers
```

Comment:
Nitpick: I am not a believer in Doxygen's "brief" and "long" summaries. I get the idea, but in practice no one wants to write two versions of an explanation. It's hard enough getting folks to write a single one. :-)

Reply:
I don't always use it, but I think for things like this it's good to give the details of how to use the function. Most of the time though, I just use the brief description if it's clear enough